task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -564,7 +565,11 @@ def run(ti: RuntimeTaskInstance, log: Logger) -> ToSupervisor | None:
                 msg = early_exit
                 return msg
 
-            result = _execute_task(context, ti)
+
+            try:
```

Comment:
This double/nested try-except looks weird

Reply:
Can we handle this at the same level as `except TaskDeferred as defer:` on L583?

Reply:
> Can we handle this at the same level as `except TaskDeferred as defer:` on L583?

I agree that it looks a bit funny - I did it to avoid duplicating the logic below the nested `try...except`. I'll wrap it in another function, and then it might look a bit better.

Reply:
Reverted